app/lib/user.server.ts

```diff
@@ -0,0 +1,8 @@
+import type { User } from '@prisma/client';
+import { setIsReconciliationNeeded } from '~/models/system-state.server';
+import { deleteUserByUsername } from '~/models/user.server';
+
+export async function deleteUser(username: User['username']) {
+  await deleteUserByUsername(username);
+  setIsReconciliationNeeded(true);
```

Comment:
`return` here, so the `Promise` is returned"

```ts
return setIsReconciliationNeeded(true);
```

Reply:
Instead of doing this, what if we add `setIsReconciliationNeeded(true);` inside `deleteUserByUsername` and just use it?